src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -449,6 +433,36 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		}
 	}
 
+	private loadContentMetadata(metadata: any): void {
+		this._savedKernelInfo = metadata.kernelspec;
+		this._defaultLanguageInfo = metadata.language_info;
+		// If language info was serialized in the notebook, attempt to use that to decrease time
+		// required until colorization occurs
+		if (this._defaultLanguageInfo) {
+			this.updateLanguageInfo(this._defaultLanguageInfo);
+		}
+		this._savedConnectionName = metadata.connection_name;
+		this._multiConnection = metadata.multi_connection ? metadata.multi_connection : false;
+
+		//Telemetry of loading notebook
+		if (metadata.azdata_notebook_guid && metadata.azdata_notebook_guid.length === 36) {
+			//Verify if it is actual GUID and then send it to the telemetry
+			let regex = new RegExp('(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}');
+			if (regex.test(metadata.azdata_notebook_guid)) {
+				this.adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.TelemetryAction.Open)
+					.withAdditionalProperties({ azdata_notebook_guid: metadata.azdata_notebook_guid })
+					.send();
+			}
+		}
+		Object.keys(metadata).forEach(key => {
+			let expectedKeys = ['kernelspec', 'language_info', 'tags', 'connection_name', 'multi_connection'];
+			// If custom metadata is defined, add to the _existingMetadata object
+			if (expectedKeys.indexOf(key) < 0) {
```

Comment:
If you keep the keyword list, then using a Set here might be better since you only care if the collection contains the specified key. Also you can initialize it as a readonly field to avoid re-creating the same collection every time we load metadata.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set